selftests/test_deproxy.py

```diff
@@ -407,5 +409,397 @@ def test_pipeline_request(self):
+        client.methods.append("GET")
+        client.wait_for_response(strict=True, n=4)
+        self.assertEqual(client.last_response.status, "200")
+        self.assertEqual(
+            [int(response.status) for response in client.responses],
+            [200, 100, 200, 200],
+            "Invalid responses sequence",
+        )
+
+    def test_request_pipeline_3_requests_empty_body(self):
+        self.disable_deproxy_auto_parser()
+
+        self.start_all_services(client=False)
+
+        server = self.get_server("deproxy")
+        server.sleep_when_receiving_data = 2
+
+        client = self.get_client("deproxy")
+        client.start()
+
+        client.make_requests(
+            requests=[
+                client.create_request(
+                    method="PUT",
+                    headers=[
+                        ("Content-Length", "9"),
+                        ("Content-Type", "application/json"),
+                    ],
+                    uri="/test",
+                    body="3" * 9,
+                    version="HTTP/1.1",
+                ),
+                client.create_request(
+                    method="PUT",
+                    headers=[
+                        ("Expect", "100-continue"),
+                        ("Content-Length", "0"),
+                        ("Content-Type", "application/json"),
+                    ],
+                    uri="/expect",
+                    version="HTTP/1.1",
+                ),
+                client.create_request(
+                    method="GET",
+                    headers=[],
+                    uri="/test",
+                    version="HTTP/1.1",
+                ),
+            ],
+            pipelined=True,
+        )
+        client.wait_for_response(timeout=10)
+
+        self.assertEqual(
+            [int(response.status) for response in client.responses],
+            [200, 200, 200],
+            "Invalid responses sequence",
+        )
+
+    def test_request_pipeline_3_requests_encoding_chunked(self):
+        self.disable_deproxy_auto_parser()
+
+        self.start_all_services(client=False)
+
+        client = self.get_client("deproxy")
+        client.start()
+
+        client.make_requests(
+            requests=[
+                client.create_request(
+                    method="PUT",
+                    headers=[
+                        ("Content-Length", "9"),
+                        ("Content-Type", "application/json"),
+                    ],
+                    uri="/test",
+                    body="3" * 9,
+                    version="HTTP/1.1",
+                ),
+                client.create_request(
+                    method="PUT",
+                    headers=[
+                        ("Expect", "100-continue"),
+                        ("Content-Length", "100"),
+                        ("Content-Type", "text/plain"),
+                        ("Content-Encoding", "chunked"),
```

Comment:
In this tests we should use `Transfer-Encoding: chunked`, not content encoding. Also we need to build chunked body for this request